### DISCUSSION TITLE
log4j vulnerability?

### DISCUSSION THREAD

**Discussion by halon314** [immense/Remotely#437] [Q&A]
Is there any chance Remotely could be affect by the log4j vulnerability?  i'm not familiar with it enough to know if Remotely using any of the tools that lead to that vulnerability.

**Comment by stevesobol** [comment]
The probability of the log4j vulnerability affecting Remotely is approximately 0.00%.

Remotely is written in C# (and, if I recall correctly, C). log4j is the most popular logging framework in the Java universe, and there _is_ a log4net project which, like log4j, is under the Apache Foundation's stewardship... but even if Remotely used log4net, I've heard nothing about it having the same problems log4j had that led to the CVEs.

That said... I just took a quick look at the .csproj files for each of the projects, and as of today, there is only one project that doesn't use Microsoft's logging framework, and that project provides its own simple logging mechanism, so the answer to your question would be "definitely not." 

Hope that helps. Have a great day :)

**Comment by halon314** [comment]
Well, that's good peace of mind. I thank you for your time and the confirmation that we're safe from that vulnerability.
Have a great day!
